lib/charms/layer/apache_zeppelin.py

```diff
@@ -13,15 +24,37 @@ class Zeppelin(object):
 
     :param DistConfig dist_config: The configuration container object needed.
     """
+    @classmethod
+    def get(cls):
+        """
+        Dynamically instantiate this or a subclass, to allow other layers
+        to override portions of this implementation.
+        """
+        impl = layer.options('apache-zeppelin')['implementation_class']
+        module_name, cls_name = impl.rsplit('.', 1)
+        cls = getattr(import_module(module_name), cls_name)
+        return cls()
```

Comment:
This doesn't feel Pythonic. I'd expect another layer to just subclass this class, and override what they want. Is there a reason for them not to do so?